api/base.py

```
@@ -85,18 +85,13 @@ def __init__(self, call_interval):
         with self.lock:
-            if random_time:
-                wait_time = random.uniform(random_min, random_max)
-                time.sleep(wait_time)
             now = time.time()
-            time_elapsed = now - self.last_call
-            if time_elapsed <= self.call_interval:
-                time.sleep(self.call_interval - time_elapsed)
-                self.last_call = time.time()
-                return
+            base_wait = max(self.last_call + self.call_interval - now, 0)
+            extra_wait = random.uniform(random_min, random_max) if random_time else 0
```

Ruff (0.15.5)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)